### DISCUSSION TITLE
Build a java client by maven or gradle.

### DISCUSSION THREAD

**Discussion by jainvikas8** [aws4embeddedlinux/meta-aws-demos#37] [General]
Hi @thomas-roos,

    Is there any classes/recipe/layers that could be used to build a provider/java client and the resulting jar file can be used by https://github.com/aws4embeddedlinux/meta-aws/tree/master/recipes-devtools/amazon-corretto for Greengrass which can be integrated as part of distro image? 

There is maven, gradle, but only find https://github.com/intel-iot-devkit/meta-iot-cloud/blob/master/recipes-support/maven/maven_3.6.3.bb which seems to be old.

   Any thoughts, please?

Thanks.

**Comment by nateglims** [comment]
I have looked into this previously and the first big hurdle is that Maven requires Maven to build, so a binary recipe or bootstrapping from an older ANT built version is required. The maintainers of meta-java may have more insight.

For now, our recommendation (and team use) has been building java projects outside of Yocto and providing them as a binary during a build.

**Reply by jainvikas8** [reply]
Thanks, Will raise a discussion on meta-java to see if they have any thoughts on this.

Yes, building the binary is an option, and downloading it and installing it in the distro. But was looking into if there was any explored option here.

**Reply by jainvikas8** [reply]
https://github.com/meta-java/meta-java/discussions/28
